PR: https://github.com/odoo/odoo/pull/37185

From: b16ce3f89e50aa8589e878bca8378f2dbac3f3f6
From: qmo-odoo
Date: 2019-09-20 14:22:06

Structural Changes: 3
Total Changes: 43

[IMP] mass_mailing_sms: add missing SMS information to UTM campaigns

This commit adds the concept of sms into the utm.campaign

  * Add "Send new sms" button to the cta
  * Add a SMS stat button
  * Add a sms page in the notebook

LINKS

Task ID 2074813 (FP request UTM refactor)

closes odoo/odoo#37185

Pr: #37185
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/mass_mailing_sms/models/__init__.py
+++ b/addons/mass_mailing_sms/models/__init__.py
@@ -7,3 +7,4 @@ from . import mailing_list
 from . import mailing_mailing
 from . import mailing_trace
 from . import sms_sms
+from . import utm

--- a/None
+++ b/addons/mass_mailing_sms/models/utm.py
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class UtmCampaign(models.Model):
+    _inherit = 'utm.campaign'
+
+    mailing_sms_ids = fields.One2many(
+        'mailing.mailing', 'campaign_id',
+        domain=[('mailing_type', '=', 'sms')],
+        string='Mass SMS')
+    mailing_sms_count = fields.Integer('Number of Mass SMS', compute="_compute_mailing_sms_count")
+
+    @api.depends('mailing_sms_ids')
+    def _compute_mailing_sms_count(self):
+        for campaign in self:
+            campaign.mailing_sms_count = len(campaign.mailing_sms_ids)
+
+    def action_create_mass_sms(self):
+        action = self.env.ref('mass_mailing.action_create_mass_mailings_from_campaign').read()[0]
+        action['context'] = {
+            'default_campaign_id': self.id,
+            'default_mailing_type': 'sms',
+            'search_default_assigned_to_me': 1,
+            'search_default_campaign_id': self.id,
+            'default_user_id': self.env.user.id,
+        }
+        return action
+
+    def action_redirect_to_mailing_sms(self):
+        action = self.env.ref('mass_mailing.action_view_mass_mailings_from_campaign').read()[0]
+        action['context'] = {
+            'default_campaign_id': self.id,
+            'default_mailing_type': 'sms',
+            'search_default_assigned_to_me': 1,
+            'search_default_campaign_id': self.id,
+            'default_user_id': self.env.user.id,
+        }
+        action['domain'] = [('mailing_type', '=', 'sms')]
+        return action
